trio/_core/_multierror.py

```diff
@@ -361,95 +391,7 @@ def concat_tb(head, tail):
     return current_head
 
 
-################################################################
-# MultiError traceback formatting
-#
-# What follows is terrible, terrible monkey patching of
-# traceback.TracebackException to add support for handling
-# MultiErrors
-################################################################
-
-traceback_exception_original_init = traceback.TracebackException.__init__
-
-
-def traceback_exception_init(
-    self,
-    exc_type,
-    exc_value,
-    exc_traceback,
-    *,
-    limit=None,
-    lookup_lines=True,
-    capture_locals=False,
-    compact=False,
-    _seen=None,
-    **kwargs,
-):
-    if sys.version_info >= (3, 10):
-        kwargs["compact"] = compact
-
-    # Capture the original exception and its cause and context as TracebackExceptions
-    traceback_exception_original_init(
-        self,
-        exc_type,
-        exc_value,
-        exc_traceback,
-        limit=limit,
-        lookup_lines=lookup_lines,
-        capture_locals=capture_locals,
-        _seen=_seen,
-        **kwargs,
-    )
-
-    seen_was_none = _seen is None
-
-    if _seen is None:
-        _seen = set()
-
-    # Capture each of the exceptions in the MultiError along with each of their causes and contexts
-    if isinstance(exc_value, MultiError):
-        embedded = []
-        for exc in exc_value.exceptions:
-            if id(exc) not in _seen:
-                embedded.append(
-                    traceback.TracebackException.from_exception(
-                        exc,
-                        limit=limit,
-                        lookup_lines=lookup_lines,
-                        capture_locals=capture_locals,
-                        # copy the set of _seen exceptions so that duplicates
-                        # shared between sub-exceptions are not omitted
-                        _seen=None if seen_was_none else set(_seen),
-                        **kwargs,
-                    )
-                )
-        self.embedded = embedded
-    else:
-        self.embedded = []
-
-
-traceback.TracebackException.__init__ = traceback_exception_init  # type: ignore
-traceback_exception_original_format = traceback.TracebackException.format
-
-
-def traceback_exception_format(self, *, chain=True):
-    yield from traceback_exception_original_format(self, chain=chain)
-
-    for i, exc in enumerate(self.embedded):
-        yield "\nDetails of embedded exception {}:\n\n".format(i + 1)
-        yield from (textwrap.indent(line, " " * 2) for line in exc.format(chain=chain))
-
-
-traceback.TracebackException.format = traceback_exception_format  # type: ignore
-
-
-def trio_excepthook(etype, value, tb):
-    for chunk in traceback.format_exception(etype, value, tb):
-        sys.stderr.write(chunk)
-
-
-monkeypatched_or_warned = False
-
+# Remove when IPython gains support for exception groups
```

Comment:
Could link to https://github.com/ipython/ipython/issues/13753

Reply:
Done.